src/pipeline/modules/segmentation.py

```diff
@@ -0,0 +1,38 @@
+from modules.module import Module
+from tensorflow import keras
+from .data import Data
+from .types import Modules
+import numpy as np
+import cv2
+
+"""
+Perform semantic segmentation using the DeepLabV3+ model.
+"""
+class SemanticSegmentation(Module):
+
+    def __init__(self, data: Data, input: any):
+        super(SemanticSegmentation, self).__init__("Semantic Segmentation DeepLabv3+", 
+            Modules.SEGMENTATION, data)
+        self.paths = input # paths to model atrifacts 
+
+    """
+    Perform inference using the images given.
+    Each image should adhere to specific dimensions in order to be
+    used by the model.
+    The model returns a mask for each of the predefined labels that can be
+    used to highlight areas of interest.
+
+    Args:
+        img: the input image(s)
+        rest: non-specific module arguments
+    """
+    def run(self, data: Data):
+        self.prepare(data)
+        model = keras.models.load_model(self.paths[data.input[0].channels])
+        predictions = []
+        for image in data.modules[Modules.PREPROCESS]["clipping"]:
+            predictions.append(model.predict(np.expand_dims((image.get()), axis=0)))
```

Comment:
Isn't `axis` 0 by default?

Reply:
There is no default argument for the axis parameter.